diff --git a/Makefile b/Makefile
index ff76e85599..df57c4a7fa 100644
--- a/Makefile
+++ b/Makefile
@@ -1700,6 +1700,13 @@ endif
 	$(build)=$(build-dir) $(@:.ko=.o)
 	$(Q)$(MAKE) -f $(srctree)/scripts/Makefile.modpost
 
+quiet_cmd_genenv = GENENV $@
+cmd_genenv = $(OBJCOPY) --dump-section .rodata.default_environment=$@ env/common.o; \
+	sed --in-place -e 's/\x00/\x0A/g' $@
+
+u-boot-initial-env: u-boot.bin
+	$(call if_changed,genenv)
+	
 # FIXME Should go into a make.lib or something
 # ===========================================================================
 
